scripts/data.py

```diff
@@ -0,0 +1,216 @@
+    return False
+
+
+def log(text: str, indent: int):
+    """Log to console output."""
+    click.echo(click.wrap_text(text, initial_indent=(indent * " ")))
+
+
+@click.group()
+def cli():
+    """Script that helps with bootstrapping example data."""
+    pass
+
+
+@cli.command("populate")
+@click.argument("address", nargs=1, required=False)
+def populate(address: str):
+    """Populate arangodb with example data."""
+    global server_address
+    log_tabstop = 4
+    log_indent = 0
+
+    if address is not None:
+        address_parts = address.split(":")
+        address_parts = [x for x in address_parts if x]
+
+        if len(address_parts) != 2:
+            raise Exception("Invalid passed address.")
+
+        server_address = address
+
+    log(f"Populating data on {server_address}...", indent=log_indent)
+
+    for path in DATA_DIR.iterdir():
+        workspace = path.name
+
+        log(f'Processing dataset "{workspace}"', indent=log_indent)
+        log_indent += log_tabstop
+
+        files = tuple(path.glob("*.csv"))
+
+        try:
+            if check_workspace_exists(workspace):
+                log(
+                    f'Workspace "{workspace}" already exists, skipping...',
+                    indent=log_indent,
+                )
+                continue
+        except requests.exceptions.ConnectionError:
+            raise Exception(f"\tConnection could not be established at {address}")
+
+        if not create_workspace(workspace):
+            log(f"Error creating workspace {workspace}.", indent=log_indent)
+            continue
+
+        # Create Tables
+        for file in files:
+            table_name = file.stem
+
+            if table_exists(workspace, table_name):
+                log(
+                    f'Table "{table_name}" already exists, skipping...',
+                    indent=log_indent,
+                )
+
+            else:
+                with file.open(mode="r") as csv_file:
+                    csv_data = csv_file.read()
+
+                if not create_table(workspace, table_name, csv_data):
+                    log(f"Error creating table {table_name}.", indent=log_indent)
+                else:
+                    log(f"Table {table_name} created.", indent=log_indent)
+
+        # Create Graphs
+        edge_tables = get_edge_tables(workspace)
+        log(f"Generating graphs...", indent=log_indent)
+
+        for edge_table in edge_tables:
+            rows = get_table_rows(workspace, edge_table)
+            rows = [
+                {k: v for k, v in row.items() if k == "_from" or k == "_to"}
+                for row in rows
+            ]
+
+            associated_node_tables = set()
+
+            for row in rows:
+                associated_node_tables.add(row["_from"].split("/")[0])
+                associated_node_tables.add(row["_to"].split("/")[0])
+
+            create_graph(
+                workspace,
+                f"{'-'.join([edge_table, *associated_node_tables])}",
```

Comment:
Can we instead have the graph named the same as the workspace? That will give us better standard names for the graphs, rather than having to name them by the tables, which will inevitably lead to confusion.

Reply:
The reason that I did this is to allow for the creation of multiple graphs if that's desired. However, if that's not a realistic use case then I'd be happy to make he change.

Reply:
Let's go with the simple case where we have one graph per dataset. As soon as we have a more complex dataset, we can (and will have to) revisit this.